docs/content/preview/yugabyte-platform/configure-yugabyte-platform/set-up-cloud-provider/on-premises.md

```diff
@@ -750,6 +750,130 @@ As an alternative to setting crontab permissions, you can install systemd-specif
+           • Node Agent Registration Successful
+   You can install a systemd service on linux machines by running node-agent-installer.sh -t install-service (Requires sudo access).
+   ```
+
+4. Run the following command to enable the node agent as a systemd service, which is required for self-upgrade and other functions: 
+
+   ```sh
+   sudo node-agent-installer.sh -t install-service  
+   ```
+
+
+### Registration
+
+To enable secured communication, the node agent is automatically registered during its installation so the YugabyteDB Anywhere is aware of its existence. You can also register and unregister the node agent manually during configuration.
+
+The YugabyteDB Anywhere API provides the endpoint via the following API: 
+
+```
+POST /api/v1/customers/<customer_id>/node_agents
+```
+
+The following is the node agent registration command:
+
+```sh
+node-agent node register --api-token <api_token>
+```
+
+If you need to overwrite any previously configured values, you can use the following parameters within the registration command:
+
+- `--node_ip` represents the node IP address.
+- `--url` represents the YugabyteDB Anywhere address.
+
+For secured communication, YugabyteDB Anywhere generates a key pair (private, public, and server certificate) that is sent to the node agent as part of its registration process.
+
+You can obtain a list of existing node agents using the following API: 
+
+```http
+GET /api/v1/customers/<customer_id>/node_agents
+```
+
+To unregister a node agent, use the following API: 
+
+```http
+DELETE /api/v1/customers/<customer_id>/node_agents/<node_agent_id>
+```
+
+### Operations
+
+A node agent supports the following commands:
+
+- `node-agent node register` is used for registering a node and node agent to YugabyteDB Anywhere. This is needed before starting a node agent server. YugabyteDB Anywhere can maintain a table for node agents.
+- `node-agent service start` and `node-agent service stop` are used for starting or stopping the node agent as a gRPC server.
+- `node-agent node preflight-check` is used for checking if a node is configured as a YugabyteDB Anywhere node. After the node agent and the node have been registered with YugabyteDB Anywhere, this command can be run on its own, if the result needs to be published to YugabyteDB Anywhere. For more information, see [Preflight check](#preflight-check).
+- `node-agent node unregister` is used for unregistersing the node and node agent from YugabyteDB Anywhere. This can be done to restart the registration process.
+
+#### Preflight check
+
+Once the node agent is installed, configured, and connected to YugabyteDB Anywhere, you can perform a series of preflight checks without sudo privileges by using the following command:
+
+```sh
+node-agent node preflight-check
+```
+
+The result is generated, uploaded to the existing node instance creation API, and posted in a tabular form on the terminal. Expect an output similar to the following:
```

Comment:
The result of the checks is sent to YugabyteDB Anywhere for validation and validated information is posted in a tabular form on the terminal. As long as there is a failure against a required check, you can fix and re-run the preflight-check to verify it.

Reply:
Changed